### ISSUE TITLE
vaccine_choices, v_min and v_max in Auspice config

### ISSUE THREAD

**Issue by trvrb** [nextstrain/augur#407]
@jameshadfield ---

Take a look at seasonal flu config files. These currently use `vmin` and `vmax` for `colorings`: 

* https://github.com/nextstrain/seasonal-flu/blob/master/config/auspice_config_h3n2_fitness.json#L41

And use `vaccine_choices` as a base level attribute:

* https://github.com/nextstrain/seasonal-flu/blob/master/config/auspice_config_h3n2_fitness.json#L103

With the new validation of config files implemented in d74f649082ee3b4c0df9e4405b09e1847fd757ce these are both erroring out:

```
Validating config file config/auspice_config_h3n2.json against the JSON schema
	ERROR: Additional properties are not allowed ('vaccine_choices' was unexpected). Trace: additionalProperties
	ERROR: Additional properties are not allowed ('vmax', 'vmin' were unexpected). Trace: properties - colorings - items - additionalProperties
Validation of config/auspice_config_h3n2.json failed. Please check the formatting of this file & refer to the augur documentation for further help.
```

I'm assuming that `vaccine_choices` is still correct, but that `vmin` and `vmax` are deprecated? I don't see `vmin` and `vmax` here: https://github.com/nextstrain/augur/blob/v6/augur/data/schema-export-v2.json. And I it looks like `vaccine_choices` makes it into the exported JSON as a `vaccine` attribute on a node?

**Comment by jameshadfield** [comment]
`vmin` and `vmax` aren't used in modular augur / auspice (they seem to have been used quite some time ago since they are exported from prepare/process augur scripts). Note that LBI has been hardcoded in auspice for ~2 years -- see https://github.com/nextstrain/auspice/issues/805. The new schema is deliberately designed allow extensions such as these (it was a comment in an early version of the schema), but this isn't implemented in augur or auspice yet. They should be removed from the seasonal flu build. 

re: `vaccine_choices` this seems to be a shortcoming of `augur export v2` where the code looks like this 😱 
```python
    def _transfer_vaccine_info(node, raw_data):
        pass
```

**Comment by trvrb** [comment]
Okay. Thanks James. I'll drop `vmin` and `vmax` but leave `vaccine_choices`. I guess one thought here while we're working on this. `vaccine_choices` are in the form:

```
  "vaccine_choices": {
    "B/Malaysia/2506/2004": "2006-09-25",
    "B/Brisbane/60/2008": "2009-09-25",
    "B/Colorado/6/2017": "2018-02-22",
    "B/Washington/2/2019": "2019-09-27"
  }
```

But `vaccine` in nodes have `serum`, `selection_date`, `start_date`, `end_date`. Like `colorings` and `maintainers` might be good to make this an array of objects and be explicit about these fields.

**Comment by jameshadfield** [comment]
Yes. The auspice JSON schema looks like

```json
                        "vaccine": {
                            "description": "Vaccine information",
                            "properties": {
                                "serum": {
                                    "description": "strain used to raise sera (for ???)",
                                    "$comment": "Currently in the flu & dengue trees",
                                    "type": "boolean"
                                },
                                "selection_date": {
                                    "description": "Vaccine selection date",
                                    "$comment": "this is currently in the metadata JSON",
                                    "type": "string",
                                    "pattern": "^[0-9X]{4}-[0-9X]{2}-[0-9X]{2}$"
                                },
                                "start_date": {
                                    "description": "Vaccine usage start date",
                                    "type": "string",
                                    "pattern": "^[0-9X]{4}-[0-9X]{2}-[0-9X]{2}$"
                                },
                                "end_date": {
                                    "description": "When the vaccine was stopped",
                                    "$comment": "if vaccine still in use, don't include this property",
                                    "type": "string",
                                    "pattern": "^[0-9X]{4}-[0-9X]{2}-[0-9X]{2}$"
                                }
                            }
                        },
```
So should this be mirrored in the auspice config, or is there a better way to get this information in (including sera etc)? I would think it's more akin to a metadata trait somewhere.

**Comment by jameshadfield** [comment]
Yes -- the auspice exported JSONs have quite a rich variety of vaccine related info, most of it currently unused by auspice. 

Auspice, when viewing v1 JSONs, transfers the `vaccine_choices` to the corresponding `node.node_attrs.vaccine.selection_date` properties. 

In terms of how this information is exported from augur, do you think it should still be defined in the config file, bearing in mind that we will one day hopefully have more rich information, or should it be part of the metadata?

**Comment by trvrb** [comment]
I would either suggest keeping `vaccine_choices` in `auspice_config.json` and adding structure:
```
  "vaccine_choices": [
    {
     "strain": "B/Malaysia/2506/2004",
     "selection_date": "2006-09-25",
     "start_date": "2007-06-01",
     "end_date": "2010-05-31"
     },
...
```

Or putting this information in a traits JSON:
```
  "nodes": {
    "B/Malaysia/2506/2004": {
     "vaccine_selection_date": "2006-09-25",
     "vaccine_start_date": "2007-06-01",
     "vaccine_end_date": "2010-05-31"  
```

This is similar to how `author` works, where export looks for magic words (like `title` and `journal`) and uses them to construct the `author` object.

@rneher, @emmahodcroft: Preferences?

Sidenote: The use of `serum` as `node.node_attrs.vaccine.serum` is extraneous with the node strain that is being labeled. I'd recommend dropping it.

**Comment by jameshadfield** [comment]
PR #334 closes this issue. After talking with @trvrb today we decided a node-data JSON for vaccines was appropriate, and allows us to add in more information about each vaccine strain in the future. You can see the seasonal-flu build using these here: https://github.com/nextstrain/seasonal-flu/pull/40

**Comment by trvrb** [comment]
This issue has been resolved in the `v6` branch by PR #410.
